pkg/router/template/router.go

```diff
@@ -553,6 +555,10 @@ func (r *templateRouter) shouldWriteCerts(cfg *ServiceAliasConfig) bool {
 		if hasRequiredEdgeCerts(cfg) {
 			return true
 		} else {
+			if cfg.TLSTermination == routeapi.TLSTerminationReencrypt && hasReencryptDestinationCACert(cfg) {
```

Comment:
Is there a reason to leave this in the else clause vs put it in its own 'else if'?